shopinvader/migrations/14.0.4.0.0/post-migrate.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    _logger.info("Install module shopinvader_auth_api_key")
+    if not version:
+        return
+
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    module = env["ir.module.module"].search(
```

Comment:
question: as this runs after the module is updated will existing `auth_api_key_id` relations be kept?
Shall we collect them on pre-migration and restore here?

Reply:
AFAIK columns are never removed by the ORM. Therefore, the relation should be kept.

Reply:
@simahawk I will do a test

Reply:
@simahawk Starting with the 14.0 branch, I created a shopinvader backend in an empty db with a new api key. I then switched to my dev branch and upgraded shopinvader... shopinvader_auth_api_key was automatically installed and I didn't lose any data on my existing backend (my api key is still there). Therefore, all seems OK with the migration script.

Reply:
thanks for testing and for the feedback! TIL :wink:

Reply:
I'm not sure I understand why this worked, but :+1: